### PR TITLE
Renamed Step curve to Threshold.

### PR DESCRIPTION
**Pull request by dragostis**
In the interest of using the name in the case of Material steppers,
this commit renames Step curve to Threshold.

### CODE REVIEW

File: packages/flutter/lib/src/animation/curves.dart [hunk 1]
```diff
@@ -112,11 +112,11 @@ class Interval extends Curve {
 }
 
 /// A curve that is 0.0 until it hits the threshold, then it jumps to 1.0.
-class Step extends Curve {
-  /// Creates a step cruve.
+class Threshold extends Curve {
+  /// Creates a threshold cruve.
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/curves.dart]: please fix the spelling of "curve" while you're here. :-)

File: packages/flutter/test/animation/curves_test.dart [hunk 2]
```diff
@@ -26,7 +26,7 @@ void main() {
   });
 
   test('Step has a step', () {
```
Comment (Hixie) [inline packages/flutter/test/animation/curves_test.dart]: rename the test and the variables in the test

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/animation/curves_test.dart (hunk 2 above)
Comment (dragostis) [inline packages/flutter/test/animation/curves_test.dart]: Good point!

**Pull request merged**